src/pow.cpp

```diff
@@ -428,6 +432,8 @@ RCP<const Basic> pow_expand(const RCP<const Pow> &self)
         }
     }
     RCP<const Basic> result = Add::from_dict(add_overall_coeff, std::move(rd));
+    RCP<const Number> i = integer(-1);
+    if(negative_pow) result = rcp(new Pow(result, i));
```

Comment:
Thinking about this more, I don't think it can trigger assert errors, because the result of expand() is always `Add` I think. What about a case like this one: `1/((x+y)^2-2*x*y-y^2)`?